core/lib/utils/mail/mail_address.dart

```diff
@@ -416,6 +510,30 @@ class MailAddress with EquatableMixin {
     return pos;
   }
 
+  static int _parseLocalPartDetails(StringBuffer localPartSB, String address, int pos) {
+    StringBuffer localPartDetailsSB = StringBuffer();
+
+    while (true) {
+      if (pos >= address.length) {
+        break;
+      }
+      var postChar = address[pos];
+      if (postChar == '@') {
+        // End of local-part-details
+        break;
+      } else {
+        localPartDetailsSB.write(postChar);
+        pos++;
+      }
+    }
+
+    String localPartDetails = localPartDetailsSB.toString();
+    // should validate localPartDetails
```

Comment:
IMO, it doesn't make sense

Reply:
i am meaning to come back to this comment: currently any string will be accepted as a localPartDetails, but the string should follow the same rules as the name of a folder, for example `user+folder%name@domain.com` cannot be valid since no folder can be named `folder%name`